pkg/manager/config/manager.go

```diff
@@ -106,16 +158,7 @@ func (e *ConfigManager) del(ctx context.Context, ns, key string) error {
 }
 
 func (e *ConfigManager) Close() error {
-	if err := e.etcdClient.Close(); err != nil {
-		e.logger.Error("fail to close config manager", zap.Error(err))
-		return err
-	}
-	return nil
-}
-
-func appendSlashToDirPath(dir string) string {
-	if len(dir) == 0 || dir[len(dir)-1] == '/' {
-		return dir
-	}
-	return dir + "/"
+	e.cancel()
+	e.wg.Wait()
+	return errors.Wrapf(e.etcdClient.Close(), "fail to close config manager")
```

Comment:
e.etcdClient is never closed

Reply:
Why?

Reply:
Because it needs to release resources. See what it does:
```go
func (c *Client) Close() error {
	c.cancel()
	if c.Watcher != nil {
		c.Watcher.Close()
	}
	if c.Lease != nil {
		c.Lease.Close()
	}
	if c.conn != nil {
		return toErr(c.ctx, c.conn.Close())
	}
	return c.ctx.Err()
}
```

Reply:
But there is `errors.Wrapf(e.etcdClient.Close(), ...)`?

Reply:
Oops, I missed it.